wrangler-core/src/main/java/co/cask/directives/date/FormatDate.java

```diff
@@ -76,21 +78,32 @@ public void destroy() {
     for (Row row : rows) {
       Row dt = new Row(row);
       int idx = dt.find(column);
-      if (idx != -1) {
-        Object object = row.getValue(idx);
-        if (object != null && object instanceof Date) {
-          dt.setValue(idx, destinationFmt.format((Date) object));
-        } else {
+
+      if (idx == -1) {
+        throw new DirectiveExecutionException(toString() + " : '" + column + "' column is not defined in the row. " +
+                                                "Please check the wrangling step."
+        );
+      }
+
+      Object object = row.getValue(idx);
+
+      if (object != null) {
+        try {
+          if (object instanceof LocalDate) {
+            object = ((LocalDate) object).atStartOfDay(ZoneId.systemDefault());
+          } else if (object instanceof LocalTime) {
+            object = ZonedDateTime.of(LocalDate.of(1970, 1, 1), (LocalTime) object, ZoneId.systemDefault());
+          }
+
+          dt.setValue(idx, destinationFmt.format((ZonedDateTime) object));
+        } catch (ClassCastException e) {
```

Comment:
instead of catching a ClassCastException, remove the try-catch and throw this exception in the else clause where you are checking the object class. Using exceptions to control the flow of a program is brittle and makes everything difficult to read.